hw/bsp/dialog_da1469x-dk-pro/da1469x_header_tool.py

```diff
@@ -0,0 +1,282 @@
+                                            self.ivt[:8], b'\xaa\x33', 64,
+                                            b'\x00'*64)
+
+            sig_bytes =  struct.pack('<BB8s2sH64s', *sig_section)
+
+            # signature is computed over device administration section, padding,
+            # and encrypted image
+            if not self.revocation_list:
+                admin_section = device_administration_section(b'\xaa\x44', 0)
+                admin_bytes = struct.pack('<2sH', *admin_section)
+            else:
+                revoke_section = revocation_section(b'\xaa\x55',
+                                                len(self.revocation_list * 2))
+                revoke_bytes = struct.pack('<2sH', *revoke_section)
+                for r in self.revocation_list:
+                    revoke_bytes += struct.pack('BB', *r)
+
+                admin_section = device_administration_section(b'\xaa\x44',
+                                                             len(revoke_bytes))
+                admin_bytes = struct.pack('<2sH', *admin_section)
+                admin_bytes += revoke_bytes
+
+
+            # pad remainder from header to beginning of binary
+            padlen = 1024 - len(sec_bytes) - len(admin_bytes) - len(sig_bytes) - len(fw_hdr_bytes)
+            pad_bytes = b'\xff' * padlen
+
+            signature = self.sig_key.sign_digest(admin_bytes + pad_bytes +
+                                                 self.data)
+
+            sig_section = signature_section(self.sig_slot, self.decrypt_slot,
+                                            self.ivt[:8], b'\xaa\x33', 64,
+                                            signature)
+            sig_bytes =  struct.pack('<BB8s2sH64s', *sig_section)
+            self.img += fw_hdr_bytes + sec_bytes + sig_bytes + admin_bytes + pad_bytes + self.data
+
+        else:
+            crc32 = binascii.crc32(self.data)
+            header =  fw_header(b'Qq', self.img_size, crc32, self.version,
+                                    int(time.time()), 0x400)
+            fw_hdr_bytes = struct.pack('<2sII16sII', *header)
+            sec_section = security_section(b'\xaa\x22', 0)
+            sec_bytes = struct.pack('<2sH', *sec_section)
+            padlen = 1024 - len(sec_bytes) - len(fw_hdr_bytes)
+            pad_bytes = b'\xff' * padlen
+            self.img += fw_hdr_bytes + sec_bytes + pad_bytes + self.data
+
+
+    def dump_to_file(self, path):
+        with open(path, "wb") as f:
+            f.write(self.img)
+
+def validate_slot_index(ctx, param, value):
+    if not value or value in range(8):
+        return value
+    else:
+        raise click.BadParameter("Slot value has to be between 0 and 7")
+
+def build_revocation_list(ctx, param, value):
+    revocation_list = []
+    lookup_table = {'s': 0xa1, 'd': 0xa2, 'u': 0xa3}
+    pattern = re.compile("s|u|d:[0-7]$")
+    for entry in value:
+        t,s = entry.split(':')
```

Comment:
Shouldn't this line be moved below the check for bad parameter below?

Reply:
yeah good catch